server/master/src/main/java/org/apache/accumulo/master/tableOps/tableImport/PopulateMetadataTable.java

```diff
@@ -66,28 +66,29 @@
     this.tableInfo = ti;
   }
 
-  static Map<String,String> readMappingFile(VolumeManager fs, ImportedTableInfo tableInfo)
-      throws Exception {
-
+  static void readMappingFile(VolumeManager fs, ImportedTableInfo tableInfo, String importDir,
+      Map<String,String> fileNameMappings) throws Exception {
     try (BufferedReader in = new BufferedReader(
-        new InputStreamReader(fs.open(new Path(tableInfo.importDir, "mappings.txt")), UTF_8))) {
-      Map<String,String> map = new HashMap<>();
-
-      String line = null;
+        new InputStreamReader(fs.open(new Path(importDir, "mappings.txt")), UTF_8))) {
+      String line, prev;
       while ((line = in.readLine()) != null) {
         String[] sa = line.split(":", 2);
-        map.put(sa[0], sa[1]);
-      }
+        prev = fileNameMappings.put(sa[0], importDir + "/" + sa[1]);
```

Comment:
Doing `importDir + "/" + sa[1]` repeatedly has the potential to use a lot more memory than just `sa[1]`.  Maybe this could be an issue for really large imports.  This could be addressed with a value object that points to importDir.  Something like `fileNameMappings.put(sa[0], new Pair<>(importDir, sa[1]))`.

Reply:
Was considering a value object here. Also considering just writing the mappings.txt file with importDir a part of the mapping itself.

Reply:
I am not sure if anything needs to be changed, it was just something I thought of when looking at the code.   I was thinking of the case of a really long import dir and lots of files, seems like that case could eat up lots of memory.